src/number_theory/basic.lean

```diff
@@ -41,4 +50,247 @@ begin
+      aux₂ hp' hp1, ← ring_hom.map_nat_cast (mk I), ← ring_hom.map_pow, ← ring_hom.map_mul,
+      eq_zero_iff_mem, mem_span_singleton, pow_two, mul_dvd_mul_iff_left hp.1],
+    intro h,
+    exact hxy₂ (dvd_mul_of_dvd_right (hp.dvd_of_dvd_pow h) _) }
+end
+
+lemma pow_prime_sub_pow_prime :
+  multiplicity ↑p (x ^ p - y ^ p) = multiplicity ↑p (x - y) + 1 :=
+by rw [← geom_sum₂_mul, multiplicity.mul hp,
+    geom_sum₂_eq_one hp hp1 hxy₁ hxy₂, add_comm]
+
+lemma pow_prime_pow_sub_pow_prime_pow (a : ℕ) :
+  multiplicity ↑p (x ^ p ^ a - y ^ p ^ a) = multiplicity ↑p (x - y) + a :=
+begin
+  induction a with a h_ind,
+  { rw [enat.coe_zero, add_zero, pow_zero, pow_one, pow_one] },
+  rw [← nat.add_one, enat.coe_add, enat.coe_one, ← add_assoc,
+    ← h_ind, pow_succ', pow_mul, pow_mul],
+  apply pow_prime_sub_pow_prime hp hp1,
+  { rw ← geom_sum₂_mul,
+    exact dvd_mul_of_dvd_right hxy₁ _ },
+  { rw ← mul_pow,
+    intro h,
+    exact hxy₂ (hp.dvd_of_dvd_pow h) }
+end
+
+end integral_domain
+
+section lifting_the_exponent
+
+variables {p : ℕ} (hp : p.prime) (hp1 : p % 2 = 1)
+include hp hp1
+
+/-- **Lifting the exponent lemma** for odd primes -/
+theorem int.pow_sub_pow {x y : ℤ} (hxy₁ : ↑p ∣ x - y) (hxy₂ : ¬↑p ∣ x * y) (n : ℕ) :
+  multiplicity ↑p (x ^ n - y ^ n) = multiplicity ↑p (x - y) + multiplicity p n :=
+begin
+  cases n,
+  { rw [multiplicity.zero, add_top, pow_zero, pow_zero, sub_self, multiplicity.zero] },
+  have h : (multiplicity _ _).dom := finite_nat_iff.mpr ⟨hp.ne_one, n.succ_pos⟩,
+  rcases eq_some_iff.mp (enat.coe_get h).symm with ⟨⟨k, hk⟩, hpn⟩,
+  conv_lhs { rw [hk, pow_mul, pow_mul] },
+  rw nat.prime_iff_prime_int at hp,
+  rw ← int.nat_cast_eq_coe_nat at *,
+  rw [pow_sub_pow' hp, pow_prime_pow_sub_pow_prime_pow hp hp1 hxy₁ hxy₂, enat.coe_get],
+  { rw ← geom_sum₂_mul,
+    exact dvd_mul_of_dvd_right hxy₁ _ },
+  { rw ← mul_pow,
+    intro h,
+    exact hxy₂ (hp.dvd_of_dvd_pow h) },
+  { iterate 2 { rw int.nat_cast_eq_coe_nat },
+    rw int.coe_nat_dvd,
+    rintro ⟨c, rfl⟩,
+    refine hpn ⟨c, _⟩,
+    rw [pow_succ', mul_assoc, ← hk] }
+end
+
+theorem int.pow_add_pow {x y : ℤ} (hxy₁ : ↑p ∣ x + y) (hxy₂ : ¬↑p ∣ x * y)
+  {n : ℕ} (hn : odd n) :
+  multiplicity ↑p (x ^ n + y ^ n) = multiplicity ↑p (x + y) + multiplicity p n :=
+begin
+  rw ← sub_neg_eq_add at hxy₁,
+  rw [← dvd_neg, neg_mul_eq_mul_neg] at hxy₂,
+  rw [← sub_neg_eq_add, ← sub_neg_eq_add, ← nat.neg_pow_of_odd hn],
+  exact int.pow_sub_pow hp hp1 hxy₁ hxy₂ n
+end
+
+theorem nat.pow_sub_pow {x y : ℕ} (hxy : y ≤ x)
+  (hxy₁ : p ∣ x - y) (hxy₂ : ¬p ∣ x * y) (n : ℕ) :
+  multiplicity p (x ^ n - y ^ n) = multiplicity p (x - y) + multiplicity p n :=
+begin
+  iterate 2 { rw ← int.coe_nat_multiplicity },
+  rw [int.coe_nat_sub (nat.pow_le_pow_of_le_left hxy n),
+    int.coe_nat_pow, int.coe_nat_pow],
+  rw ← int.coe_nat_dvd at hxy₁ hxy₂,
+  rw int.coe_nat_sub hxy at hxy₁ ⊢,
+  rw int.coe_nat_mul at hxy₂,
+  exact int.pow_sub_pow hp hp1 hxy₁ hxy₂ n
```

Comment:
Another way to write this, emphasising that it's a specialisation of `int.pow_sub_pow`, is:
```suggestion
  have hpow := int.coe_nat_sub (nat.pow_le_pow_of_le_left hxy n),
  have hsub := int.coe_nat_sub hxy,
  convert int.pow_sub_pow hp hp1 _ _ n using 1,
  { simp only [← int.coe_nat_multiplicity, hpow, int.coe_nat_pow] },
  { simp only [← int.coe_nat_multiplicity, hsub, int.coe_nat_pow] },
  { simpa only [hsub] using int.coe_nat_dvd.mpr hxy₁ },
  { simpa only [int.coe_nat_mul] using mt int.coe_nat_dvd.mp hxy₂ }
```

Reply:
Also, I'd rather not use `convert` for these ones